Src/WitsmlExplorer.Frontend/components/ContentViews/Charts/LogsGraph.tsx

```diff
@@ -0,0 +1,307 @@
+  height: number;
+  type: string;
+  width: number;
+  x: number;
+  y: number;
+}
+
+interface LogItem {
+  name: string;
+  start: number;
+  end: number;
+  uid: string;
+}
+
+export const LogsGraph = (): React.ReactElement => {
+  const dimItemIndex = 0;
+  const dimStart = 1;
+  const dimEnd = 2;
+  const cartesianXBounds = [];
+  const cartesianYBounds = [];
+  const data: DataItem[] = [];
+  const categories: number[] = [];
+  const { navigationState } = useContext(NavigationContext);
+  const { selectedWellbore, selectedLogTypeGroup } = navigationState;
+
+  // dedicated colors for items containing specific names
+  const reservedColours: Map<string, string> = new Map([
+    ["#0000FF", "surface"],
+    ["#FF0000", "downhole"],
+    ["#FFFF00", "memory"]
+  ]);
+
+  const {
+    operationState: { timeZone }
+  } = useContext(OperationContext);
+  const [logs, setLogs] = useState<LogObject[]>([]);
+  const [resetCheckedItems, setResetCheckedItems] = useState(false);
+
+  useEffect(() => {
+    if (selectedWellbore?.logs) {
+      setLogs(selectedWellbore.logs.filter((log) => calculateLogTypeId(selectedWellbore, log.indexType) === selectedLogTypeGroup));
+    }
+  }, [selectedLogTypeGroup, selectedWellbore]);
+
+  const isTimeIndexed = () => {
+    return selectedLogTypeGroup === calculateLogTypeTimeId(selectedWellbore);
+  };
+
+  const getTableData = (): LogItem[] => {
+    return logs.map((log) => {
+      return {
+        name: log.name,
+        start: selectedWellbore && isTimeIndexed() ? +new Date(formatDateString(log.startIndex, timeZone)) : log.startIndex === null ? 0 : +log.startIndex?.replace("m", ""),
+        end: selectedWellbore && isTimeIndexed() ? +new Date(formatDateString(log.endIndex, timeZone)) : log.endIndex === null ? 0 : +log.endIndex?.replace("m", ""),
+        uid: log.uid,
+        category: null
+      };
+    });
+  };
+
+  const sortedLogs = getTableData().sort((a, b) => {
+    return a.start - b.start;
+  });
+
+  // calculates equal vertical distribution
+  // needs to be more tuned
+  const verticalAxisZoomMaxValue = () => {
+    const numberOfCategories = categories.length;
+    if (numberOfCategories < 13) {
+      return 100;
+    }
+    const a = 0.00358858;
+    const b = 1.02584;
+    const c = 70.3405;
+    const result = a * Math.pow(numberOfCategories, 2) - b * numberOfCategories + c;
+    return result;
+  };
+
+  const gridMaxHeight = () => {
+    const numberOfCategories = categories.length;
+    if (numberOfCategories < 13) {
+      return numberOfCategories * 50;
+    }
+    return 600;
+  };
+
+  const randomColor = (name: string): string => {
+    let result = "";
+    reservedColours.forEach(function (value, key) {
+      if (name.includes(value)) {
+        result = key;
+      }
+    });
+    // returns reserved colours
+    if (result !== "") {
+      return result;
+    }
+    return generateColor(name);
+  };
+
+  const generateColor = (name: string): string => {
```

Comment:
The random colors make the graph look a bit chaotic. Should the same color be used for all "other" logs, or maybe the color palette should be narrowed down? Maybe discuss it with Asbjørn.

Reply:
Discussion with Asbjørn is needed.